hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestNMClient.java

```diff
@@ -125,576 +117,362 @@ public void postTransition(
-  private void waitForContainerTransitionCount(
-      Container container,
-      org.apache.hadoop.yarn.server.nodemanager.
-          containermanager.container.ContainerState state, long transitions)
-      throws YarnException {
-    long transitionCount = -1;
-    do {
-      if (transitionCount != -1) {
-        try {
-          Thread.sleep(10);
-        } catch (InterruptedException e) {
-          throw new YarnException(
-              "Timeout at transition count:" + transitionCount, e);
-        }
-      }
-      transitionCount = DebugSumContainerStateListener
-          .getTransitionCounter(container.getId(), state);
-    } while (transitionCount != transitions);
+  private void testContainerStatusCompleted(
+      Container container, String... diagnostics
+  ) throws YarnException, IOException {
+    ContainerStatus actualStatus = nmClient
+        .getContainerStatus(container.getId(), container.getNodeId());
+    while (ContainerState.COMPLETE != actualStatus.getState()) {
+      sleep(100);
+      actualStatus = nmClient.getContainerStatus(container.getId(), container.getNodeId());
+    }
+    assertEquals(container.getId(), actualStatus.getContainerId());
+    assertTrue(Arrays.asList(
+        ContainerExitStatus.KILLED_BY_APPMASTER,
+        ContainerExitStatus.SUCCESS
+    ).contains(actualStatus.getExitStatus()));
+    for (String diagnostic : diagnostics) {
+      assertTrue(actualStatus.getDiagnostics().contains(diagnostic));
+    }
   }
 
-  private void sleep(int sleepTime) {
-    try {
-      Thread.sleep(sleepTime);
-    } catch (InterruptedException e) {
-      e.printStackTrace();
-    }
+  @SuppressWarnings("deprecation")
+  private void testIncreaseContainerResource(Container container) {
+    assertYarnException(
+        () -> nmClient.increaseContainerResource(container),
+        container.getId() + " has update version ");
   }
 
-  private void testGetContainerStatus(Container container, int index,
-      ContainerState state, String diagnostics, List<Integer> exitStatuses)
-          throws YarnException, IOException {
-    while (true) {
-      sleep(250);
-      ContainerStatus status = nmClient.getContainerStatus(
-          container.getId(), container.getNodeId());
-      // NodeManager may still need some time to get the stable
-      // container status
-      if (status.getState() == state) {
-        assertEquals(container.getId(), status.getContainerId());
-        assertTrue("" + index + ": " + status.getDiagnostics(),
-            status.getDiagnostics().contains(diagnostics));
-
-        assertTrue("Exit Statuses are supposed to be in: " + exitStatuses +
-                ", but the actual exit status code is: " +
-                status.getExitStatus(),
-            exitStatuses.contains(status.getExitStatus()));
-        break;
-      }
-    }
+  private void testRestartContainer(Container container) throws IOException, YarnException {
+    nmClient.restartContainer(container.getId());
```

Comment:
What's the reason for completely removing this idiom across many test<ContainerEvent> methods?
I'm talking about: 
```
if (!e.getMessage().contains(
          "can only be changed when a container is in RUNNING state")) {
        throw (AssertionError)
            (new AssertionError("Exception is not expected: " + e)
                .initCause(e));

```
Methods including: 
- testContainer
- testIncreaseContainerResource
- testRestartContainer
- testRollbackContainer
- testCommitContainer
- testReInitializeContainer

Reply:
Sure,
In short: this test is a mess
In long, those exceptions was never thrown